node/impl/full/mpool.go

```diff
@@ -117,7 +116,7 @@ func (a *MpoolAPI) MpoolPushMessage(ctx context.Context, msg *types.Message) (*t
 		if err != nil {
 			return nil, xerrors.Errorf("estimating gas used: %w", err)
 		}
-		msg.GasLimit = int64(float64(gasLimit) * GasMargin)
+		msg.GasLimit = int64(float64(gasLimit) * a.Mpool.GetConfig().GasLimitOverestimation)
```

Comment:
it might be a little slow to call GetConfig() repeatedly, because it copies the actual config object.

Reply:
It is only on pushing the message. It is slow already (as it executes the message for GasLimit estimation).

Reply:
Fair enough! I am fine with it as is, but maybe we should do it inside the message pool instead?